server/src/main/java/org/apache/druid/metadata/SqlSegmentsMetadataQuery.java

```diff
@@ -834,6 +861,21 @@ private static int computeNumChangedSegments(List<String> segmentIds, int[] segm
     return numChangedSegments;
   }
 
+  private static void appendConditionForVersions(
+      final StringBuilder sb,
+      final List<String> versions
+  )
+  {
+    if (CollectionUtils.isNullOrEmpty(versions)) {
```

Comment:
Nit: Better check this condition in the caller (even if it is being done in two places) and call this method only when needed. Also, rename this method to `getConditionForVersions` and return a `String` from this method rather than passing the builder.

Reply:
Updated to `getConditionForVersions()`. There is a similar existing method `appendConditionForIntervalsAndMatchMode()` that can perhaps be changed later to follow the updated pattern.